### PR TITLE
chore(CLI): Move destroy logic from docker to envd engine.

### LINKED ISSUE #1046: chore(CLI): Move destroy logic from docker to envd engine.
## Description

Now the func `Destroy(ctx context.Context, name string) (string, error)` is defined in `pkg/docker/docker.go`

We should move it to pkg/envd/engine.go and implement in pkg/envd/docker.go

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by yvestumushimire**
Fixes #1046

### CODE REVIEW

File: e2e/cli/up_test.go [hunk 1]
```diff
@@ -37,10 +34,6 @@ var _ = Describe("up command", Ordered, func() {
 		envdApp := app.New()
 		err := envdApp.Run(append(baseArgs, "bootstrap"))
 		Expect(err).NotTo(HaveOccurred())
-		cli, err := docker.NewClient(context.TODO())
```
Comment (gaocegege) [inline e2e/cli/up_test.go:40]: We need to keep the test logic here, or the test case will fail

Comment (yvestumushimire) [reply e2e/cli/up_test.go:40]: How can I replicate Destroy method in tests? I saw `dockerEngine` is not exported @gaocegege

Comment (gaocegege) [reply e2e/cli/up_test.go:40]: You can define Destroy method in env.Engine. Then fake a context

```go
	c := types.Context{Runner: types.RunnerTypeDocker}
	opt := envd.Options{
		Context: c,
	}
	envdEngine, err := envd.New(clicontext.Context, opt)
	if err != nil {
		return errors.Wrap(err, "failed to create envd engine")
	}
```

File: pkg/app/destroy.go [hunk 2]
```diff
@@ -74,13 +75,18 @@ func destroy(clicontext *cli.Context) error {
 		}
 		ctrName = filepath.Base(buildContext)
 	}
+	c := types.Context{Runner: types.RunnerTypeDocker}
```
Comment (gaocegege) [inline pkg/app/destroy.go:78]: Could you please get the context from the home manager? Maybe like this 

https://github.com/tensorchord/envd/blob/main/pkg/app/env_ls.go#L41:#L48

Comment (yvestumushimire) [reply pkg/app/destroy.go]: Thank you just changed it :)

**Review by gaocegege — COMMENTED** [review]
/lgtm
/approve

Thanks for your contribution! :tada: :+1:

**Pull request merged**